governance/pyth_staking_sdk/src/pyth-staking-client.ts

```diff
@@ -0,0 +1,410 @@
+    const mint = globalConfig.pythTokenMint;
+
+    const senderTokenAccount = await getAssociatedTokenAddress(
+      mint,
+      this.wallet.publicKey
+    );
+
+    const transactions =
+      await TransactionBuilder.batchIntoVersionedTransactions(
+        this.wallet.publicKey,
+        this.provider.connection,
+        [
+          {
+            instruction: createTransferInstruction(
+              senderTokenAccount,
+              getStakeAccountCustodyAddress(stakeAccountPositions),
+              this.wallet.publicKey,
+              amount
+            ),
+            signers: [],
+          },
+        ],
+        {}
+      );
+
+    await sendTransactions(transactions, this.connection, this.wallet);
+  }
+
+  public async withdrawTokensFromStakeAccountCustody(
+    stakeAccountPositions: PublicKey,
+    amount: bigint
+  ) {
+    const globalConfig = await this.getGlobalConfig();
+    const mint = globalConfig.pythTokenMint;
+
+    const receiverTokenAccount = await getAssociatedTokenAddress(
+      mint,
+      this.wallet.publicKey
+    );
+
+    const instruction = await this.stakingProgram.methods
+      .withdrawStake(new BN(amount.toString()))
+      .accounts({
+        destination: receiverTokenAccount,
+        stakeAccountPositions,
+      })
+      .instruction();
+
+    const transactions =
+      await TransactionBuilder.batchIntoVersionedTransactions(
+        this.wallet.publicKey,
+        this.provider.connection,
+        [
+          {
+            instruction,
+            signers: [],
+          },
+        ],
+        {}
+      );
+
+    await sendTransactions(transactions, this.connection, this.wallet);
+  }
+
+  public async stakeToPublisher(options: {
+    stakeAccountPositions: PublicKey;
+    publisher: PublicKey;
+    amount: bigint;
+  }) {
+    const { stakeAccountPositions, publisher, amount } = options;
+    this.integrityPoolProgram.methods
+      .delegate(new BN(amount.toString()))
+      .accounts({
+        owner: this.wallet.publicKey,
+        publisher,
+        stakeAccountPositions,
+      })
+      .rpc();
+  }
+
+  public async getUnlockSchedule(options: {
+    stakeAccountPositions: PublicKey;
+  }) {
+    const { stakeAccountPositions } = options;
+    const stakeAccountMetadataAddress = getStakeAccountMetadataAddress(
+      stakeAccountPositions
+    );
+    const stakeAccountMetadata =
+      await this.stakingProgram.account.stakeAccountMetadataV2.fetch(
+        stakeAccountMetadataAddress
+      );
+    const vestingSchedule = convertBNToBigInt(stakeAccountMetadata.lock);
+
+    const config = await this.getGlobalConfig();
+
+    return getUnlockSchedule({
+      vestingSchedule,
+      pythTokenListTime: config.pythTokenListTime!,
+    });
+  }
+
+  public async advanceDelegationRecord(options: {
+    stakeAccountPositions: PublicKey;
+  }) {
+    // TODO: optimize to only send transactions for publishers that have positive rewards
```

Comment:
agree

Reply:
we have to fix it soon. Currently there are 8 transactions which is probably not acceptable.